Extension/src/LanguageServer/settings.ts

```diff
@@ -178,43 +180,51 @@ class Settings {
 
     protected get Section(): vscode.WorkspaceConfiguration { return this.settings; }
 
-    protected getWithFallback<T>(section: string, deprecatedSection: string): T {
-        const info: any = this.settings.inspect<T>(section);
+    // If the setting has an undefined default, look for the workspaceFolder, workspace and global values as well.
+    public getArrayOfStringsWithUndefinedDefault(section: string): string[] | undefined;
+    public getArrayOfStringsWithUndefinedDefault(section: string, allowNull: boolean): string[] | undefined | null;
+    public getArrayOfStringsWithUndefinedDefault(section: string, allowNull: boolean = false): string[] | undefined | null {
+        const info: any = this.settings.inspect<string[]>(section);
+        if (allowNull && info === null) {
+            return null;
```

Comment:
Is it correct to check if `info` is null? I thought it would check if `info.workspaceFolderValue`, etc was null.